### PR TITLE
EXAMPLES/NBX: re-implemented examples using NBX API

### PR DESCRIPTION
**Pull request by hoopoepg**
- updated examples using send/recv NBX API

### CODE REVIEW

File: examples/ucp_client_server.c [hunk 1]
```diff
@@ -268,20 +269,24 @@ static int send_recv_stream(ucp_worker_h ucp_worker, ucp_ep_h ep, int is_server,
                             int current_iter)
 {
     char recv_message[TEST_STRING_LEN]= "";
+    ucp_request_param_t param;
     test_req_t *request;
     size_t length;
 
+    param.op_attr_mask = UCP_OP_ATTR_FIELD_CALLBACK | UCP_OP_ATTR_FIELD_DATATYPE;
```
Comment (yosefe) [inline examples/ucp_client_server.c:276]: pls change the example to use user_data to point to &complete flag, instead of extending request size

Comment (hoopoepg) [reply examples/ucp_client_server.c]: done... hello_world example update too?

File: examples/ucp_client_server.c [hunk 2]
```diff
@@ -297,21 +311,28 @@ static int send_recv_tag(ucp_worker_h ucp_worker, ucp_ep_h ep, int is_server,
...
+    ucp_request_param_t param;
+    void *request;
+    test_req_t ctx;
+
+    ctx.complete = 0;
+    param.op_attr_mask = UCP_OP_ATTR_FIELD_CALLBACK |
+                         UCP_OP_ATTR_FIELD_DATATYPE |
+                         UCP_OP_ATTR_FIELD_USER_DATA;
+    param.datatype     = ucp_dt_make_contig(TEST_STRING_LEN);
+    param.user_data    = &ctx;
```
Comment (yosefe) [inline examples/ucp_client_server.c:323]: maybe not pass datatype, and set count=TEST_STRING_LEN

File: examples/ucp_hello_world.c [hunk 3]
```diff
@@ -207,9 +208,10 @@ static int run_ucx_client(ucp_worker_h ucp_worker)
     msg->data_len = local_addr_len;
     memcpy(msg + 1, local_addr, local_addr_len);
 
-    request = ucp_tag_send_nb(server_ep, msg, msg_len,
-                              ucp_dt_make_contig(1), tag,
-                              send_handler);
+    send_param.op_attr_mask = UCP_OP_ATTR_FIELD_CALLBACK;
+    send_param.cb.send      = send_handler;
```
Comment (yosefe) [inline examples/ucp_hello_world.c:212]: no user_data?

File: examples/ucp_hello_world.c [hunk 4]
```diff
@@ -399,9 +402,10 @@ static int run_ucx_server(ucp_worker_h ucp_worker)
     ret = generate_test_string((char *)(msg + 1), test_string_length);
     CHKERR_JUMP(ret < 0, "generate test string", err_free_mem_type_msg);
 
-    request = ucp_tag_send_nb(client_ep, msg, msg_len,
-                              ucp_dt_make_contig(1), tag,
-                              send_handler);
+    send_param.op_attr_mask = UCP_OP_ATTR_FIELD_CALLBACK;
+    send_param.cb.send      = send_handler;
```
Comment (yosefe) [inline examples/ucp_hello_world.c:406]: no user_data?

**Comment by yosefe** [comment]
@hoopoepg can you pls add stream API to the example as well? now that it's merged

File: examples/ucp_hello_world.c (hunk 3 above)
Comment (hoopoepg) [reply examples/ucp_hello_world.c:214]: added user_data

File: examples/ucp_hello_world.c (hunk 4 above)
Comment (hoopoepg) [reply examples/ucp_hello_world.c:414]: added user_data

**Comment by hoopoepg** [comment]
@yosefe ok to squash?

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
VM FS issue
bot:pipe:retest

**Pull request merged**